account.go

```diff
@@ -0,0 +1,180 @@
+package client
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+)
+
+// Get information about your account. See https://api.omg.lol/#token-get-account-retrieve-account-information
+func (c *Client) GetAccountInfo() (*Account, error) {
+	req, err := http.NewRequest("GET", fmt.Sprintf("%s/account/%s/info", c.HostURL, c.Auth.Email), nil)
+	if err != nil {
+		return nil, err
+	}
+
+	body, err := c.doRequest(req)
+	if err != nil {
+		return nil, err
+	}
+
+	var account Account
+	if err := json.Unmarshal(body, &account); err != nil {
+		fmt.Printf("Error unmarshalling response: %v\n", err)
+		return nil, err
+	}
+
+	return &account, nil
+}
+
+// Get all addresses associated with your account. See https://api.omg.lol/#token-get-account-retrieve-addresses-for-an-account
+func (c *Client) GetAccountAddresses() (*Addresses, error) {
+	req, err := http.NewRequest("GET", fmt.Sprintf("%s/account/%s/addresses", c.HostURL, c.Auth.Email), nil)
+	if err != nil {
+		return nil, err
+	}
+
+	body, err := c.doRequest(req)
+	if err != nil {
+		return nil, err
+	}
+
+	var addresses Addresses
+	err = json.Unmarshal(body, &addresses)
+	if err != nil {
+		return nil, fmt.Errorf("Error unmarshalling response: %v", err)
+	}
+
+	return &addresses, nil
+}
+
+// Get the name associated with the account. See https://api.omg.lol/#token-get-account-retrieve-the-account-name
+func (c *Client) GetAccountName() (*AccountName, error) {
+	req, err := http.NewRequest("GET", fmt.Sprintf("%s/account/%s/name", c.HostURL, c.Auth.Email), nil)
+	if err != nil {
+		return nil, err
+	}
+
+	body, err := c.doRequest(req)
+	if err != nil {
+		return nil, err
+	}
+
+	var name AccountName
+	if err := json.Unmarshal(body, &name); err != nil {
+		fmt.Printf("Error unmarshalling response: %v\n", err)
+		return nil, err
+	}
+
+	return &name, nil
+}
+
+// Set the name associated with the account. See https://api.omg.lol/#token-post-account-set-the-account-name
+func (c *Client) SetAccountName(name string) (*AccountName, error) {
+	jsonData := fmt.Sprintf(`{"name": "%s"}`, name)
```

Comment:
I'd encourage writing structs with json tags (as you have in models.go for other stuff) for everything, including request bodies